> @jasl I don't see any bits about cross compiling in the rustc invocations in the build log you provided. What commands are you using to compile the playground?
> 
> Can you start in a new shell and let share what sequence of command you run to get this error?

1 clone a freash repo

```
jasl@Jasl-MBP2021:~/Workspaces/tmp$ git clone https://github.com/artichoke/playground.git
Cloning into 'playground'...
remote: Enumerating objects: 9707, done.
remote: Counting objects: 100% (253/253), done.
remote: Compressing objects: 100% (192/192), done.
remote: Total 9707 (delta 156), reused 115 (delta 60), pack-reused 9454
Receiving objects: 100% (9707/9707), 153.53 MiB | 16.68 MiB/s, done.
Resolving deltas: 100% (6547/6547), done.
```

2 `. scripts/install-emscripten-toolchain.sh`

```
jasl@Jasl-MBP2021:~/Workspaces/tmp/playground on trunk$ . scripts/install-emscripten-toolchain.sh
Cloning into 'emsdk'...
remote: Enumerating objects: 83, done.
remote: Counting objects: 100% (83/83), done.
remote: Compressing objects: 100% (74/74), done.
remote: Total 83 (delta 7), reused 37 (delta 4), pack-reused 0
Receiving objects: 100% (83/83), 91.84 KiB | 435.00 KiB/s, done.
Resolving deltas: 100% (7/7), done.
Resolving SDK version '2.0.26' to 'sdk-releases-upstream-823d37b15d1ab61bc9ac0665ceef6951d3703842-64bit'
Installing SDK 'sdk-releases-upstream-823d37b15d1ab61bc9ac0665ceef6951d3703842-64bit'..
Installing tool 'node-14.18.2-64bit'..
Downloading: /Users/jasl/Workspaces/tmp/playground/emsdk/zips/node-v14.18.2-darwin-x64.tar.gz from https://storage.googleapis.com/webassembly/emscripten-releases-builds/deps/node-v14.18.2-darwin-x64.tar.gz, 32076686 Bytes
Unpacking '/Users/jasl/Workspaces/tmp/playground/emsdk/zips/node-v14.18.2-darwin-x64.tar.gz' to '/Users/jasl/Workspaces/tmp/playground/emsdk/node/14.18.2_64bit'
Done installing tool 'node-14.18.2-64bit'.
Installing tool 'python-3.9.2-64bit'..
Downloading: /Users/jasl/Workspaces/tmp/playground/emsdk/zips/python-3.9.2-1-macos-arm64.tar.gz from https://storage.googleapis.com/webassembly/emscripten-releases-builds/deps/python-3.9.2-1-macos-arm64.tar.gz, 31582429 Bytes
Unpacking '/Users/jasl/Workspaces/tmp/playground/emsdk/zips/python-3.9.2-1-macos-arm64.tar.gz' to '/Users/jasl/Workspaces/tmp/playground/emsdk/python/3.9.2_64bit'
Done installing tool 'python-3.9.2-64bit'.
Installing tool 'releases-upstream-823d37b15d1ab61bc9ac0665ceef6951d3703842-64bit'..
Downloading: /Users/jasl/Workspaces/tmp/playground/emsdk/zips/823d37b15d1ab61bc9ac0665ceef6951d3703842-wasm-binaries-arm64.tbz2 from https://storage.googleapis.com/webassembly/emscripten-releases-builds/mac/823d37b15d1ab61bc9ac0665ceef6951d3703842/wasm-binaries-arm64.tbz2, 164665814 Bytes
Unpacking '/Users/jasl/Workspaces/tmp/playground/emsdk/zips/823d37b15d1ab61bc9ac0665ceef6951d3703842-wasm-binaries-arm64.tbz2' to '/Users/jasl/Workspaces/tmp/playground/emsdk/upstream'
Done installing tool 'releases-upstream-823d37b15d1ab61bc9ac0665ceef6951d3703842-64bit'.
Done installing SDK 'sdk-releases-upstream-823d37b15d1ab61bc9ac0665ceef6951d3703842-64bit'.
Resolving SDK version '2.0.26' to 'sdk-releases-upstream-823d37b15d1ab61bc9ac0665ceef6951d3703842-64bit'
Setting the following tools as active:
   node-14.18.2-64bit
   python-3.9.2-64bit
   releases-upstream-823d37b15d1ab61bc9ac0665ceef6951d3703842-64bit

Next steps:
- To conveniently access emsdk tools from the command line,
  consider adding the following directories to your PATH:
    /Users/jasl/Workspaces/tmp/playground/emsdk
    /Users/jasl/Workspaces/tmp/playground/emsdk/node/14.18.2_64bit/bin
    /Users/jasl/Workspaces/tmp/playground/emsdk/upstream/emscripten
- This can be done for the current shell by running:
    source "/Users/jasl/Workspaces/tmp/playground/emsdk/emsdk_env.sh"
- Configure emsdk in your shell startup scripts by running:
    echo 'source "/Users/jasl/Workspaces/tmp/playground/emsdk/emsdk_env.sh"' >> $HOME/.zprofile
Adding directories to PATH:
PATH += /Users/jasl/Workspaces/tmp/playground/emsdk
PATH += /Users/jasl/Workspaces/tmp/playground/emsdk/upstream/emscripten
PATH += /Users/jasl/Workspaces/tmp/playground/emsdk/node/14.18.2_64bit/bin

Setting environment variables:
PATH = /Users/jasl/Workspaces/tmp/playground/emsdk:/Users/jasl/Workspaces/tmp/playground/emsdk/upstream/emscripten:/Users/jasl/Workspaces/tmp/playground/emsdk/node/14.18.2_64bit/bin:/Users/jasl/.rbenv/shims:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Users/jasl/.cargo/bin
EMSDK = /Users/jasl/Workspaces/tmp/playground/emsdk
EM_CONFIG = /Users/jasl/Workspaces/tmp/playground/emsdk/.emscripten
EMSDK_NODE = /Users/jasl/Workspaces/tmp/playground/emsdk/node/14.18.2_64bit/bin/node
EMSDK_PYTHON = /Users/jasl/Workspaces/tmp/playground/emsdk/python/3.9.2_64bit/bin/python3
SSL_CERT_FILE = /Users/jasl/Workspaces/tmp/playground/emsdk/python/3.9.2_64bit/lib/python3.9/site-packages/certifi/cacert.pem
```

3 `npm run dev:release`

```
jasl@Jasl-MBP2021:~/Workspaces/tmp/playground on trunk$ npm run dev:release
... too much outputs
[watch:release] [build:release:wasm] > ruby scripts/build-wasm.rb --release
...
[watch:release] [build:release:wasm]   error: linking with `cc` failed: exit status: 1
[watch:release] [build:release:wasm]     |
    = note: "cc" "-arch" "arm64" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.0.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.1.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.10.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.11.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.12.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.13.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.14.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.15.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.2.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.3.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.4.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.5.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.6.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.7.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.8.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.build_script_build.215381f2-cgu.9.rcgu.o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f.1dsxf8w3nz7thhrj.rcgu.o" "-L" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/deps" "-L" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libstd-3e9242c3cd2b7504.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libpanic_unwind-21fb160eabff0de9.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libobject-db30465a1eaface5.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libmemchr-e382952d2c548344.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libaddr2line-ec20cbcde7f93da6.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libgimli-9ab26dd50e2d2400.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libstd_detect-2a0541465d7990f5.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_demangle-ca5a8bf0e1c973cd.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libhashbrown-661f613758a122d7.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_std_workspace_alloc-e3fb6213fb083b7c.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libunwind-e47fb12d3d9565f1.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcfg_if-5d995d79ef7c7cbe.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/liblibc-195c03df6e48132d.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/liballoc-706d571a09b848f6.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/librustc_std_workspace_core-581e7c3ef8936945.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcore-61f7a53bba9c6667.rlib" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib/libcompiler_builtins-62ccedd146f025b3.rlib" "-lSystem" "-lresolv" "-lc" "-lm" "-liconv" "-L" "/Users/jasl/.rustup/toolchains/1.56.1-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib" "-o" "/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM/release/build/proc-macro2-6f81de82f8a5ef7f/build_script_build-6f81de82f8a5ef7f" "-Wl,-dead_strip" "-nodefaultlibs" "-s" "MODULARIZE=1" "-s" "WASM=1" "-s" "ENVIRONMENT=web" "-s" "SUPPORT_LONGJMP=1"
[watch:release] [build:release:wasm]     = note: clang: error: no such file or directory: 'MODULARIZE=1'
[watch:release] [build:release:wasm]             clang: error: no such file or directory: 'WASM=1'
[watch:release] [build:release:wasm]             clang: error: no such file or directory: 'ENVIRONMENT=web'
[watch:release] [build:release:wasm]             clang: error: no such file or directory: 'SUPPORT_LONGJMP=1'
[watch:release] [build:release:wasm]
[watch:release] [build:release:wasm]
[watch:release] [build:release:wasm]   error: failed to compile `bindgen v0.59.1`, intermediate artifacts can be found at `/var/folders/z8/v1b2v_cs7zx404tc9h0rrm_00000gn/T/cargo-installVsYlNM`
[watch:release] [build:release:wasm]
[watch:release] [build:release:wasm]   Caused by:
[watch:release] [build:release:wasm]     build failed
[watch:release] [build:release:wasm]   thread 'main' panicked at 'cargo install bindgen failed', /Users/jasl/.cargo/git/checkouts/artichoke-c12ffc01676ba375/6797c18/artichoke-backend/build.rs:337:13
[watch:release] [build:release:wasm]   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[watch:release] [build:release:wasm] npm run build:release:wasm exited with code 0
[watch:release] [build:release:app]
[watch:release] [build:release:app] > @artichokeruby/playground@0.7.0 build:release:app /Users/jasl/Workspaces/tmp/playground
[watch:release] [build:release:app] > node build.mjs --release
[watch:release] [build:release:app]
[watch:release] [build:release:app] internal/process/esm_loader.js:74
[watch:release] [build:release:app]     internalBinding('errors').triggerUncaughtException(
[watch:release] [build:release:app]                               ^
[watch:release] [build:release:app]
[watch:release] [build:release:app] Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'eta' imported from /Users/jasl/Workspaces/tmp/playground/build.mjs
[watch:release] [build:release:app]     at new NodeError (internal/errors.js:322:7)
[watch:release] [build:release:app]     at packageResolve (internal/modules/esm/resolve.js:687:9)
[watch:release] [build:release:app]     at moduleResolve (internal/modules/esm/resolve.js:728:18)
[watch:release] [build:release:app]     at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:842:11)
[watch:release] [build:release:app]     at Loader.resolve (internal/modules/esm/loader.js:89:40)
[watch:release] [build:release:app]     at Loader.getModuleJob (internal/modules/esm/loader.js:242:28)
[watch:release] [build:release:app]     at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:76:40)
[watch:release] [build:release:app]     at link (internal/modules/esm/module_job.js:75:36) {
[watch:release] [build:release:app]   code: 'ERR_MODULE_NOT_FOUND'
[watch:release] [build:release:app] }
[watch:release] [build:release:app] npm ERR! code ELIFECYCLE
[watch:release] [build:release:app] npm ERR! errno 1
[watch:release] [build:release:app] npm ERR! @artichokeruby/playground@0.7.0 build:release:app: `node build.mjs --release`
npm ERR! Exit status 1
[watch:release] [build:release:app] npm ERR!
[watch:release] [build:release:app] npm ERR! Failed at the @artichokeruby/playground@0.7.0 build:release:app script.
[watch:release] [build:release:app] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
[watch:release] [build:release:app] npm WARN Local package.json exists, but node_modules missing, did you mean to install?
[watch:release] [build:release:app]
[build:release:app] npm ERR! A complete log of this run can be found in:
[watch:release] [build:release:app] npm ERR!     /Users/jasl/.npm/_logs/2022-03-14T09_14_30_453Z-debug.log
[watch:release] [build:release:app] npm run build:release:app exited with code 1
[watch:release] npm ERR! code ELIFECYCLE
npm ERR! errno 1
[watch:release] npm ERR! @artichokeruby/playground@0.7.0 build:release: `npx concurrently --max-processes 1 "npm:build:release:wasm" "npm:build:release:app"`
npm ERR! Exit status 1
[watch:release] npm ERR!
[watch:release] npm ERR! Failed at the @artichokeruby/playground@0.7.0 build:release script.
[watch:release] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
[watch:release] npm WARN Local package.json exists, but node_modules missing, did you mean to install?
[watch:release]
[watch:release] npm ERR! A complete log of this run can be found in:
[watch:release] npm ERR!     /Users/jasl/.npm/_logs/2022-03-14T09_14_30_492Z-debug.log
```

[2022-03-14T09_14_30_492Z-debug.log](https://github.com/artichoke/playground/files/8243390/2022-03-14T09_14_30_492Z-debug.log)

--- a/README.md
+++ b/README.md
@@ -31,6 +31,8 @@ development server with:
 
 ```sh
 rustup target add wasm32-unknown-emscripten
+npm ci
+cargo install bindgen
 . scripts/install-emscripten-toolchain.sh
 npm run dev:debug # or npm run dev:release
 ```

